frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,61 @@
+import { IconColor } from '#libs/enums/enums.js';
+import {
+  useCallback,
+  useHandleClickOutside,
+  useRef,
+} from '#libs/hooks/hooks.js';
+
+import { Button } from '../components.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  children: React.ReactNode;
+  title: string;
+  reference: React.RefObject<HTMLDialogElement>;
+  showCrossIcon: boolean;
+};
+
+const Modal: React.FC<Properties> = ({
+  children,
+  title,
+  reference,
```

Comment:
It should be passed via forwardRef callback

Reply:
Please check another PRS with Modal to notify guys

Reply:
done